src/apps/ENGINE/file_service.h

```diff
@@ -88,47 +88,30 @@ class FILE_SERVICE : public VFILE_SERVICE
   public:
     FILE_SERVICE();
     ~FILE_SERVICE();
-    HANDLE _CreateFile(const char *lpFileName, uint32_t dwDesiriedAccess = GENERIC_READ,
-                       uint32_t dwShareMode = FILE_SHARE_READ, uint32_t dwCreationDisposition = OPEN_EXISTING) override;
-    void _CloseHandle(HANDLE hFile) override;
-    uint32_t _SetFilePointer(HANDLE hFile, long DistanceToMove, long *lpDistanceToMoveHigh,
-                             uint32_t dwMoveMethod) override;
-    BOOL _DeleteFile(const char *lpFileName) override;
-    BOOL _WriteFile(HANDLE hFile, const void *lpBuffer, uint32_t nNumberOfBytesToWrite,
-                    uint32_t *lpNumberOfBytesWritten) override;
-    BOOL _ReadFile(HANDLE hFile, void *lpBuffer, uint32_t nNumberOfBytesToRead, uint32_t *lpNumberOfBytesRead) override;
+    std::fstream _CreateFile(const char *filename, std::ios::openmode mode) override;
+    void _CloseFile(std::fstream &fileS) override;
+    void _SetFilePointer(std::fstream &fileS, std::streamoff off, std::ios::seekdir dir) override;
+    int _DeleteFile(const char *filename) override;
+    bool _WriteFile(std::fstream &fileS, const char *s, std::streamsize count) override;
+    bool _ReadFile(std::fstream &fileS, char *s, std::streamsize count) override;
```

Comment:
I think `s` could stay a void* to avoid all the reinterpret_casts. Same for _WriteFile